### PR TITLE
DrivAerML: 4L/320d + WD=1e-2 (transfer AirfRANS regularization)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Transfer the AirfRANS regularization breakthrough to DrivAerML. On AirfRANS, WD=1e-2 + gc=1.0 was a massive win (6.8% improvement on its own), and WD=1e-2 + 4L/256d + gc=1.0 produced a 22.3% improvement. DrivAerML is at the same winning 4L/320d architecture. Testing WD=1e-2 alone (vs default WD=1e-4) at the current best DrivAerML config to isolate the regularization effect.

On AirfRANS, WD=1e-2 works because it: (1) prevents gradient explosions from accumulating at cosine LR peaks, (2) provides clean L2 regularization that complements grad-clip, (3) reduces overfitting on the relatively small 3D CFD dataset. DrivAerML has similar characteristics: small dataset, large model, rapid cosine cycling.

## Instructions

Run exactly this command (no other changes):

```bash
cd target/icml2026
SENPAI_TIMEOUT_MINUTES=180 SENPAI_MAX_EPOCHS=9999 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 \
  --cosine-t-max 30 --no-use-ema --enable-fourier \
  --model-layers 4 --model-hidden-dim 320 --model-heads 5 \
  --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --weight-decay 1e-2 \
  --wandb_group drivaerml-4L320d-reg
```

**Only change from baseline:** `--weight-decay 1e-2` (default is 1e-4).

Report in a PR comment:
- Best `val_primary/surface_rel_l2_pct` and at which epoch
- `test_primary/surface_rel_l2_pct` at that epoch
- Total epochs completed in 180 min
- W&B run ID and link
- Grad norm behavior (stabilizing or spiking at cosine peaks?)
- Whether model was still improving at cutoff

## Baseline

- **Primary metric:** `val_primary/surface_rel_l2_pct` (lower is better)
- **Current best:** 5.027% val / 6.244% test at epoch 257
- **Best PR:** #2648 (zoro — 4L/320d/5H, T_max=30, lr=5e-4, no-EMA, Fourier, 257 epochs in 180-min)
- **External target:** <3.71%
- **Baseline W&B run:** qx7z7if3
- **Reproduce baseline:**
  ```bash
  cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 \
    --cosine-t-max 30 --no-use-ema --enable-fourier \
    --model-layers 4 --model-hidden-dim 320 --model-heads 5 \
    --epochs 999 --batch-size 1 \
    --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
    --max-train-batches 394 --max-eval-batches 200
  ```

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2752 because ADVISOR: Closing — WD=1e-2 catastrophically diverges on DrivAerML (confirmed by ray #2790: grad norms hit 231x, 14.40% best vs 4.619% baseline). Also, baseline has moved to 4L/512d making 4L/320d experiments superseded..
